lua/ttt2/libraries/fileloader.lua

```diff
@@ -0,0 +1,76 @@
+---
+-- A file loading library for a custom ttt2 file loader
+-- @author Mineotopia
+
+if SERVER then
+	AddCSLuaFile()
+end
+
+local fileFind = file.Find
+local stringRight = string.Right
+
+fileloader = fileloader or {}
+
+---
+-- Sets up files by scanning through directories and including them into the runtime.
+-- @note Has to be run on both server and client for client and shared files.
+-- @param string path The path to search in
```

Comment:
I'd personally like to be a bit more explicit here about the form of the path e.g.:
```suggestion
-- @param string path The absolute path to search in, has to end with `/`
```